packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -129,11 +129,30 @@ export interface ICluster extends IResource, ec2.IConnectable {
    */
   readonly kubectlMemory?: Size;
 
+  /**
+   * A security group to associate with the Cluster Handler's Lambdas.
+   * The Cluster Handler's Lambdas are responsible for calling AWS's EKS API.
+   *
+   * Requires `placeClusterHandlerInVpc` to be set to true.
+   *
+   * @default - No security group.
+   * @attribute
+   */
+  readonly clusterHandlerSecurityGroup?: ec2.ISecurityGroup;
+
   /**
     * An AWS Lambda layer that includes the NPM dependency `proxy-agent`.
     *
-    * If not defined, a default layer will be used.
+    * @default - If not defined, a default layer will be used.
     */
+  readonly proxyAgentLayer?: lambda.ILayerVersion;
+
+  /**
+   * Deprecated
+   *
+   * @default - a layer bundled with this module.
+   * @deprecated use `proxyAgentLayer` instead
```

Comment:
Like already mentioned. Don't think it necessary, leave as is and just add always add the proxy agent layer.